nixos/modules/services/networking/pppd.nix

```diff
@@ -0,0 +1,137 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+{
+  meta = {
+    maintainers = with maintainers; [ danderson ];
+  };
+
+  options = {
+    services.pppd = {
+      package = mkOption {
+        default = pkgs.ppp;
+        defaultText = "pkgs.ppp";
+        type = types.package;
+        description = "pppd package to use.";
+      };
+
+      peers = mkOption {
+        default = {};
+        type = types.loaOf (types.submodule (
+          { name, ... }:
+          {
+            options = {
+              name = mkOption {
+                example = "dialup";
+                type = types.str;
+                description = "Name of the PPP peer.";
+              };
+
+              enable = mkOption {
+                type = types.bool;
+                default = true;
+                example = false;
+                description = "Whether to enable this PPP peer.";
+              };
+
+              autostart = mkOption {
+                type = types.bool;
+                default = true;
+                example = false;
+                description = "Whether the PPP session is automatically started at boot time.";
+              };
+
+              config = mkOption {
```

Comment:
Could consider using a structured config. See https://github.com/NixOS/rfcs/pull/42

Reply:
pppd configurations are a bit unusual compared to more modern configs, so I'm not sure if it fits cleanly inside a structured config. For example, here's my pppd config:

```
plugin rp-pppoe.so eno1
name "redacted-isp-username"
noipdefault
persist
defaultroute
hide-password
noauth
```

I'm not sure how I would convert this to a structured config, given the irregular shape. Do you have a suggestion?

Reply:
I'm not sure, was more just something to think about

Reply:
I'll think about it some more tomorrow. In the meantime, I think it's easy to switch from a text config to structured later and deprecate this field.

Reply:
After reviewing the RFC, I believe there is no obvious mapping between pppd configs and nix. The format is just too weird and irregular.

However, per the RFC's recommendations for "unsuitable formats", I changed the type of `config` to `types.lines`, so at least specifying additional configuration works reasonably.